### PR TITLE
Create bind function

### PR DESCRIPTION
**Pull request by justinsb**
Trying a more lightweight approach to creating these functions also,
as the existing process has become quite heavy.

### CODE REVIEW

File: functions/go/bind/main.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	f := Bind{}
+
+	err := f.LoadConfig(rl.FunctionConfig)
+	if err != nil {
+		rl.Results = append(rl.Results, fn.ErrorConfigObjectResult(fmt.Errorf("functionConfig error: %w", err), rl.FunctionConfig))
+		return true, nil
+	}
+
+	if f.Object == nil {
+		return false, fmt.Errorf("`data.object` should not be empty")
```
Comment (yuwenma) [inline functions/go/bind/main.go:32]: Do you mean`fmt.Errorf("functionConfig should not be empty")`?

File: functions/go/bind/main.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	// Find the matching localconfig object
+	var bindingObject *fn.KubeObject
+	{
+		// TODO: Match GK or GVK?
+		gvk := f.Object.GroupVersionKind()
+
+		matchAnnotations := map[string]string{
+			"config.kubernetes.io/local-config": "binding",
+		}
+		matches := objects.Where(fn.HasAnnotations(matchAnnotations)).Where(fn.IsGVK(gvk.Group, gvk.Version, gvk.Kind))
```
Comment (yuwenma) [inline functions/go/bind/main.go:57]: nit: `.Where(fn.IsGVK(gvk.Group, gvk.Version, gvk.Kind))` --> `.Where(f.Object.IsGVK)`

File: functions/go/bind/pkg/meta/kinds.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+)
+
+// IsClusterScoped returns true if the specified GVK is cluster scoped, or an error if this cannot be determined.
+func IsClusterScoped(gvk schema.GroupVersionKind) (bool, error) {
+	ki := findKindInfo(gvk)
+	if ki == nil {
+		return false, fmt.Errorf("kind %v not known", gvk)
+	}
+	return ki.ClusterScoped, nil
+}
```
Comment (yuwenma) [inline functions/go/bind/pkg/meta/kinds.go:30]: FYI, the go SDK has `IsClusterScoped` `IsNamespaceScoped` functions that also relies on the hard-coded namespace list https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/go/fn/internal/namespace.go#L8. I'm happy to extend the list to include CNRM and change the IsClusterScoped to only require GVK args.

File: functions/go/bind/pkg/meta/namespaceref.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+type namespaceRef struct {
+	parentObject *fn.SubObject
+
+	name      string
+	fieldPath []string
+}
+
+func buildMetadataNamespaceReference(parentObject *fn.KubeObject) (Ref, error) {
+	namespace := parentObject.GetMap("metadata").GetString("namespace")
```
Comment (yuwenma) [inline functions/go/bind/pkg/meta/namespaceref.go:32]: `parentObject.GetNamespace()` gives the same result I think. the `GetMap` can return `nil` and then causes panic I think https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/go/fn/object.go#L962

**Comment by justinsb** [comment]
Merged as part of #923 

We'll deal with the comments here in a separate PR!

**Pull request closed without merging**